test/vintage_net_wizard/backend_test.exs

```diff
@@ -0,0 +1,51 @@
+defmodule VintageNetWizard.Backend.Test do
+  use ExUnit.Case, async: true
+
+  alias VintageNetWizard.{WiFiConfiguration, Backend}
+
+  test "Can save a WiFi configuration" do
+    :ok = Backend.reset()
+    configuration = WiFiConfiguration.new("Test Network", key_mgmt: :wpa_psk, password: "1234")
+    :ok = Backend.save(configuration)
+
+    assert [configuration] == Backend.configurations()
+  end
+
+  test "sets the priority order of configurations" do
+    :ok = Backend.reset()
+    configuration1 = WiFiConfiguration.new("Test Network", key_mgmt: :wpa_psk, password: "1234")
+    configuration2 = WiFiConfiguration.new("Test Network 2", key_mgmt: :wpa_psk, password: "1234")
+
+    :ok = Backend.save(configuration1)
+    :ok = Backend.save(configuration2)
+
+    :ok = Backend.set_priority_order(["Test Network 2", "Test Network"])
```

Comment:
It might be worth using the values in the created configurations here to prevent breakage if someone happens to change the name when they are created.

```suggestion
    :ok = Backend.set_priority_order([configuration2.ssid, configuration1.ssid])
```